crates/runtime/src/modelruntime/tract.rs

```diff
@@ -32,6 +33,33 @@ pub struct Model {
     model: Plan,
 }
 
+fn transpose_and_flatten(matrix: &[Vec<f64>]) -> Vec<f64> {
+    if matrix.is_empty() || matrix[0].is_empty() {
+        return Vec::new();
+    }
+
+    let nrows = matrix.len();
+    let ncols = matrix[0].len();
+
+    // Check for uniformity in inner vector lengths
+    for row in matrix {
+        assert!(
+            row.len() == ncols,
+            "All rows must have the same number of columns"
+        );
```

Comment:
Does this indicate a bug in our code, or the user provided the wrong data? If the latter - we don't want to panic here.